### ISSUE TITLE
Out of date React dependency in CDN preventing ReactWebChat component render 

### ISSUE THREAD

**Issue by mSigson** [microsoft/BotFramework-WebChat#2602]
## Version

**4.6.0**

## Describe the bug

We're an Ember.js shop using Microsoft Bot Framework to implement a new bot in our web app.  However, to render custom components through WebChat we've added React/ReactDOM to our build to be able to use the `ReactWebChat` component.  

We want to make sure our version of React is as up to date as possible and due to the version locks in 4.6.0 our bot won't render due to duelling versions of React (see console error).  

Current project versions:
```
"react": "^16.11.0",
"react-dom": "^16.11.0"
```

We've now had to change our CDN from 
`https://cdn.botframework.com/botframework-webchat/latest/webchat.js` to `https://cdn.botframework.com/botframework-webchat/4.5.3/webchat.js`.

![Screen Shot 2019-11-13 at 10 59 13 AM](https://user-images.githubusercontent.com/29506632/68780773-be443b80-0604-11ea-8ac3-03840b920f69.png)

## Enquiry

What are the reasons for locking React/ReactDOM in 4.6.0?  
I'd like to be able to upgrade our project version of React to stay up to date.

Is there a plan to upgrade React beyond `16.8.6` and to stay up to date as versions are released? 

## To Reproduce

1. open existing React Project (react/react-dom versions greater than `16.8.6`)
2. At botframework through cdn  `https://cdn.botframework.com/botframework-webchat/latest/webchat.js`
3. Attempt to render bot through`ReactWebChat` component

## Expected bug behavior

Open console, expect to see errors from screenshot below.
Expect bot not to render/run

![Screen Shot 2019-11-13 at 11 16 14 AM](https://user-images.githubusercontent.com/29506632/68782219-06645d80-0607-11ea-971e-36c11d2d5786.png)

**Comment by corinagum** [comment]
Please see our sample and information on React and React Hooks https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/23.a.hybrid-react-npm

This includes helpful documentation and our justifications for locking React.

**Comment by gorner** [comment]
@corinagum Thanks for that response, we understand the need to lock to a single version >=16.8 but that page doesn't explain why you've locked the CDN build to 16.8.6 when the current version is 16.11.0.

Part of our team's broader point was that, because we were pointing to the "latest" CDN URL (as suggested in the [documentation](https://github.com/microsoft/BotFramework-WebChat#how-to-use)), our webchat feature broke unexpectedly after the 4.6 release. We realize there's not really a good way to notify that new releases are forthcoming to a CDN in this circumstance but the breaking nature (for us) of 4.6 was frustrating. 

We would have preferred if the documentation had just pointed to a specific version - or better yet a minor version stream like "4.5" so we would get patches - in combination with some other method of notifying devs of a new version (e.g. `console.warn`) so they could upgrade at their pace. (Of course we know we could use the NPM package as well, but there are a couple of specific reasons related to deferred loading we ended up using the CDN for now.)

**Comment by compulim** [comment]
We are not locking on a single version (16.8.6) specifically. We support React >= 16.8.6. I will verify to make sure Web Chat 4.6.0 can be loaded in React 16.11.0.

Are you integrating Web Chat inside a React app? If yes, is there a reason not using NPM version of Web Chat but CDN?

If you are loading Web Chat inside a React app, and prefer to load it via CDN. You should expose your instance of React/ReactDOM via `window.React` and `window.ReactDOM` respectively.

Technically, as explained in the error, it is caused by either:

1. There are multiple React and React DOM loaded in the web page. These React/DOM can be of same version, but they occupy different memory space.
   - You can follow this page to diagnose the problem, https://fb.me/react-invalid-hook-call#duplicate-react
1. The web page has some code using React Hooks outside of a functional component.

For CDN version on point 1, we use this load mechanism:

1. If Web Chat detect there is a version of React loaded at `window.React` and `window.ReactDOM`, we will use that version
1. Otherwise, Web Chat will use React 16.8.6 bundled inside Web Chat CDN JS file

This load strategy is designed for web apps which prefer React from CDN (unpkg.com or UMD). Related sample, https://github.com/microsoft/BotFramework-WebChat/blob/master/samples/03.a.host-with-react/index.html

For NPM version, we do not bundle any version of React. So if you are using React 16.11.0, Web Chat use your version of React (a.k.a. `peerDependencies`). This is done thru `npm install botframework-webchat`.

**Comment by compulim** [comment]
I tested loading Web Chat 4.6.0 CDN JS file inside two environments of React:

- React 16.11.0 loaded via unpkg.com
- React 16.11.0 loaded via NPM via `create-react-app`

Both works.

**Comment by gorner** [comment]
Just to close the loop on this - our team (on which @mSigson was until very recently a member) investigated this week and discovered that this was due to another issue configuration issue that apparently was never noted until after this update.

In short (I won't go into much depth since this was mostly specific to our primarily Ember.js-based project) - React and ReactDOM were available globally, but we were including them in our code in a different way using shims that conflicted with something else and pulled in a separate ReactDOM build. And it was only after the WebChat build started using hooks that we discovered this.
